cardinal/server/server.go

```diff
@@ -46,6 +53,152 @@ const (
+	})
+
+	authorizePersonaAddressHandler := runtime.OperationHandlerFunc(func(i interface{}) (interface{}, error) {
+		return ecs.AuthorizePersonaAddressResult{Success: true}, errors.New("not implemented")
+	})
+	api.RegisterOperation("POST", "/tx/core/{txType}", coreHandler)
+	api.RegisterOperation("POST", "/tx/persona/create-persona", createPersonaHandler)
+	api.RegisterOperation("POST", "/tx/persona/authorize-persona-address", authorizePersonaAddressHandler)
+
+	return nil
+}
+
+// result struct for /query/http/endpoints
+type EndpointsResult struct {
+	TxEndpoints    []string `json:"tx_endpoints"`
+	QueryEndpoints []string `json:"query_endpoints"`
+}
+
+// register query endpoints for swagger server
+func registerReadHandlerSwagger(world *ecs.World, api *untyped.API) error {
+	//var txEndpoints []string
+	coreHandler := runtime.OperationHandlerFunc(func(params interface{}) (interface{}, error) {
+		return struct {
+			test string
+		}{test: "test"}, errors.New("not implemented")
+	})
+	listHandler := runtime.OperationHandlerFunc(func(params interface{}) (interface{}, error) {
+		txs, err := world.ListTransactions()
+		txEndpoints := make([]string, 0, len(txs))
+		if err != nil {
+			return nil, err
+		}
+		for _, tx := range txs {
+			if tx.Name() != "create-persona" && tx.Name() != "authorize-persona-address" {
+				txEndpoints = append(txEndpoints, "/tx/core/"+tx.Name())
+			} else {
+				txEndpoints = append(txEndpoints, "/tx/persona/"+tx.Name())
+			}
+		}
+
+		queryEndpoints := make([]string, 0, len(world.ListReads())+3)
+		for _, read := range world.ListReads() {
+			queryEndpoints = append(queryEndpoints, "/query/core/"+read.Name())
+		}
+		queryEndpoints = append(queryEndpoints, "/query/http/endpoints")
+		queryEndpoints = append(queryEndpoints, "/query/persona/signer")
+		queryEndpoints = append(queryEndpoints, "/query/receipt/submit")
+		return EndpointsResult{
+			TxEndpoints:    txEndpoints,
+			QueryEndpoints: queryEndpoints,
+		}, nil
+	})
+	personaHandler := runtime.OperationHandlerFunc(func(i interface{}) (interface{}, error) {
+		return ReadPersonaSignerResponse{Status: "", SignerAddress: ""}, errors.New("not implemented")
+	})
+	receiptsHandler := runtime.OperationHandlerFunc(func(i interface{}) (interface{}, error) {
+		return struct {
+			start_tick uint64
+			end_tick   uint64
+			Receipt    []struct {
+				tx_hash string
+				tick    uint64
+				result  interface{}
+				errors  []string
+			}
+		}{start_tick: 0, end_tick: 0, Receipt: []struct {
+			tx_hash string
+			tick    uint64
+			result  interface{}
+			errors  []string
+		}{}}, errors.New("not implemented")
+	})
+	api.RegisterOperation("GET", "/query/core/{readType}", coreHandler)
+	api.RegisterOperation("GET", "/query/http/endpoints", listHandler)
+	api.RegisterOperation("GET", "/query/persona/signer", personaHandler)
+	api.RegisterOperation("GET", "/query/receipts/submit", receiptsHandler)
```

Comment:
I don't like this name. This endpoint takes in some world tick and returns a list of transactions between the given tick and the current tick. how about...
`/quer/receipt/list`
or
`/query/receipt/since`

'submit' seems to generic.